src/createServer.js

```diff
@@ -1,8 +1,58 @@
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
+const http = require('http');
+const { serverError } = require('./errors');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', async (req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    if (req.method === 'POST' && url.pathname === '/add-expense') {
+      try {
+        const chunks = [];
+
+        for await (const chunk of req) {
+          chunks.push(chunk);
+        }
+
+        const text = Buffer.concat(chunks).toString();
+        const data = JSON.parse(text);
+        const { date, title, amount } = data;
+
+        if (!date || !title || !amount) {
+          res.statusCode = 400;
+
+          return res.end('All fields must be filled');
```

Comment:
There's a typo in the error message. It should be more descriptive, such as 'All fields (date, title, amount) must be filled'.